### PR TITLE
BUDE-634: Building out a new time selector

### PR DESCRIPTION
**Pull request by lancempoe**
## PR Checklist

We are building out a new time selector. This functionally works now and has been reviewed with UX.

Storybook can be viewed [here](https://docspot.adnxs.net/projects/lucid/BUDE-634/?path=/docs/controls-timeselect--twelve-hour)

- Manually tested across supported browsers

  - [x] Chrome
  - [x] Firefox
  - [x] Safari

- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/components/SingleSelect/SingleSelect.tsx [hunk 1]
```diff
@@ -126,6 +126,8 @@ export interface ISingleSelectProps extends StandardProps {
 
 	maxMenuHeight?: number | string;
 
+	showIcon?: boolean;
```
Comment (lancempoe) [inline src/components/SingleSelect/SingleSelect.tsx:129]: We need a way to show and hide this icon

File: src/components/TimeSelect/TimeSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	/** Set to true to disable the TimeSelect */
+	isDisabled?: boolean;
+}
+
+enum MeridiemType {
+	AM = 'AM',
+	PM = 'PM',
+}
+
+const TimeSelect = ({
```
Comment (lancempoe) [inline src/components/TimeSelect/TimeSelect.tsx:30]: This is the new component

**Review by dreidn — APPROVED** [review]

**Review by yasskin — APPROVED** [review]
Looks good. Clean and clear code. Like the test coverage.

**Review by yasskin — APPROVED** [review]
Looks good.

**Pull request merged**